Downsample/Downsample.cpp

```diff
@@ -138,17 +138,26 @@ Eigen::MatrixXf gridSum(const Eigen::MatrixXf &r, int downsamplingFactor)
 
 Eigen::MatrixXf lineSum(const Eigen::MatrixXf &matrix, int downsamplingFactor)
 {
+    Eigen::Map<Eigen::MatrixXf> reshapedMatrixMap(const_cast<float *>(matrix.data()),
```

Comment:
This `const_cast` is safe because we don't actually alter the data, is that it?

Reply:
Yes, the input matrix to this funciton is not changed. A new one is created from it.

Reply:
Does `matrix.data()` actually get copied? Or is that data pointer just being wrapped in a new data structure, and the new structure keeps pointing to the old data?

Reply:
I don't know. How to check that?

Reply:
You have to read the Eigen docs to figure that out.

Reply:
I think the map is only mapping a existing array, specifying the alignment and the stride. Normally, modifying the `Map` will also modify the `matrix`. But I think we should do the rest of the review live, as @knatten suggested.